hack/extract.go

```diff
@@ -0,0 +1,253 @@
+/*
+Copyright 2019 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/golang-collections/collections/stack"
+)
+
+var funcs map[string]struct{}
+var funcStack *stack.Stack
+var translations map[string]interface{}
+
+func main() {
```

Comment:
AST parsing looks fun! Do you mind adding a test which runs this against a directory containing an example test .go file? I want to make sure that we can safely make improvements to this in the future without breaking edge cases.